SwrveConversationSDK/Conversation/SwrveBaseConversation.m

```diff
@@ -49,21 +52,60 @@ +(UIStoryboard*) loadStoryboard
 
 -(BOOL)assetsReady:(NSSet*)assets {
     for (SwrveConversationPane* page in self.pages) {
+
+        // check font and images from content
         for (SwrveContentItem* contentItem in page.content) {
-            if([contentItem.type isEqualToString:kSwrveContentTypeImage]) {
-                if([assets containsObject:contentItem.value]) {
-                    return true;
-                }
-                else {
+            if ([contentItem isKindOfClass:[SwrveContentImage class]]) {
+                if(![assets containsObject:contentItem.value]) {
                     DebugLog(@"Conversation asset not yet downloaded: %@", contentItem.value);
                     return false;
                 }
+            } else if ([contentItem isKindOfClass:[SwrveContentHTML class]] || [contentItem isKindOfClass:[SwrveContentStarRating class]]) {
+                if(![self isFontAsset:contentItem.style inCache: assets]) {
+                    DebugLog(@"Conversation asset not yet downloaded: %@", contentItem.style);
+                    return false;
+                }
+            } else if ([contentItem isKindOfClass:[SwrveInputMultiValue class]]) {
+                SwrveInputMultiValue *inputMultiValue = (SwrveInputMultiValue*) contentItem;
+                if(![self isFontAsset:inputMultiValue.style inCache: assets]) {
+                    DebugLog(@"Conversation asset not yet downloaded: %@", inputMultiValue.style);
+                    return false;
+                }
+
+                for (NSDictionary *value in inputMultiValue.values) {
+                    NSDictionary *style = [value objectForKey:@"style"];
+                    if(![self isFontAsset:style inCache: assets]) {
+                        DebugLog(@"Conversation asset not yet downloaded: %@", style);
+                        return false;
+                    }
+                }
+            }
+        }
+
+        // check font asset in button
+        for (SwrveConversationButton *button in page.controls) {
+            if(![self isFontAsset:button.style inCache: assets]) {
+                DebugLog(@"Conversation asset not yet downloaded: %@", button.style);
+                return false;
             }
         }
     }
+
     return true;
 }
 
+
+- (BOOL)isFontAsset:(NSDictionary *)style inCache:(NSSet *)assets {
+    BOOL isFontAssetInCache = NO;
+    if (style && [style objectForKey:@"font_file"]) {
+        NSString *fontFile = [style objectForKey:@"font_file"];
+        if (fontFile && [assets containsObject:fontFile]) {
+            isFontAssetInCache = YES;
```

Comment:
return from the end of a method is a far far better style of coding. Easier is read and less buggy.